pennylane/interfaces/tf.py

```diff
@@ -170,15 +160,14 @@ def grad(grad_output, **tfkwargs):
             grad_input = tf.matmul(grad_output_row, jacobian)
             grad_input = tf.reshape(grad_input, [-1])
 
-            if non_diff_indices:
-                # TensorFlow requires we return a gradient of size (num_variables,)
-                res = np.zeros([qnode.num_variables])
-                indices = np.fromiter(diff_indices, dtype=np.int64)
-                res[indices] = grad_input
-                grad_input = tf.constant(res, dtype=dtype)
-
```

Comment:
How come this conversion is no longer needed?

Reply:
I realised that Ville had originally taken into account non-differentiable parameters in the QNode if you leave `wrt=None`, so this was a bit of code duplication.

Two things are happening now:

* Now only differentiable args are wrapped as variables on the forward pass, so `qnode.Jacobian(wrt=None)` will only differentiate these arguments (not _all_ differentiable arguments).

* There is existing code in `qnode.Jacobian(wrt=None)` such that non-Variable arguments simply result in a '0' element in the output Jacobian.